### ISSUE TITLE
Regression : Wrong parsing for clippath

### ISSUE THREAD

**Issue by Sharcoux** [jhy/jsoup#920]
I don't get why, but with the new 1.10.3 version, if I try to parse:

```
<svg width="120" height="120" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <clipPath id="clip1">
      <circle cx="30" cy="30" r="20"/>
    </clipPath>
    <clipPath id="clip2">
      <circle cx="70" cy="70" r="20"/>
    </clipPath>
  </defs>

  <rect x="10" y="10" width="100" height="100" clip-path="url(#clip1)"/>
  <rect x="10" y="10" width="100" height="100" clip-path="url(#clip2)"/>
</svg>
```

I get:

```
<svg width="120" height="120" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <clippath id="clip1">
      <circle cx="30" cy="30" r="20" />
    
    <clippath id="clip2">
      <circle cx="70" cy="70" r="20" />
    
  </clippath></clippath></defs>

  <rect x="10" y="10" width="100" height="100" clip-path="url(#clip1)" />
  <rect x="10" y="10" width="100" height="100" clip-path="url(#clip2)" />
</svg>
```

As you may notice, the clippath siblings are getting nested...

I can't say why, but I know that it was working fine on version 1.9.2.

**Comment by krystiangorecki** [comment]
Remember you're not parsing HTML. You should parse it using [xmlParser](https://jsoup.org/apidocs/org/jsoup/parser/Parser.html#xmlParser--)
```
import java.io.IOException;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.parser.Parser;

public class JsoupIssue920 {

    public static void main(String[] args) throws IOException {
        String svgString = "<svg width=\"120\" height=\"120\" xmlns=\"http://www.w3.org/2000/svg\">"+
            "  <defs>"+
            "    <clipPath id=\"clip1\">"+
            "      <circle cx=\"30\" cy=\"30\" r=\"20\"/>"+
            "    </clipPath>"+
            "    <clipPath id=\"clip2\">"+
            "      <circle cx=\"70\" cy=\"70\" r=\"20\"/>"+
            "    </clipPath>"+
            "  </defs>"+
            ""+
            "  <rect x=\"10\" y=\"10\" width=\"100\" height=\"100\" clip-path=\"url(#clip1)\"/>"+
            "  <rect x=\"10\" y=\"10\" width=\"100\" height=\"100\" clip-path=\"url(#clip2)\"/>"+
            "</svg>";
        Document document = Jsoup.parse(svgString, "", Parser.xmlParser());
        System.out.println(document);
    }

}
```
then my output in 1.10.3 is exactly what you expect:
```
<svg width="120" height="120" xmlns="http://www.w3.org/2000/svg"> 
 <defs> 
  <clipPath id="clip1"> 
   <circle cx="30" cy="30" r="20" /> 
  </clipPath> 
  <clipPath id="clip2"> 
   <circle cx="70" cy="70" r="20" /> 
  </clipPath> 
 </defs> 
 <rect x="10" y="10" width="100" height="100" clip-path="url(#clip1)" /> 
 <rect x="10" y="10" width="100" height="100" clip-path="url(#clip2)" />
</svg>

**Comment by Sharcoux** [comment]
Note that SVG is allowed directly into HTML with HTML 5, so the HTML parser should be able to parse it correctly...

**Comment by krystiangorecki** [comment]
Ok, but second half of my answer stands true. 
A workaround for this bug while using HTML parser would be to use it with `preserveCase` setting like this:
```
		Parser parser = Parser.htmlParser();
		parser.settings(ParseSettings.preserveCase);
		Document document = Jsoup.parse(html, "", parser);
```

**Comment by krystiangorecki** [comment]
Good news. I debugged and investigated this bug and found out it's already fixed in https://github.com/jhy/jsoup/commit/397a0caeb374c55b8dcb58e09d0faebb6e017252#diff-04b5aee85909ef32bf910f9fc7d1e462R763
The fix will be included in version 1.11.x.
If you need it now you can build your own jar :)
```
git clone --depth=50 git://github.com/jhy/jsoup.git jhy/jsoup
cd jhy/jsoup
mvn package
```

**Comment by Sharcoux** [comment]
Oh great ! Well done. Thanks for all your work !
